### ISSUE TITLE
Ansible gce - create or terminate GCE instances metadata is not working as expected.

### ISSUE THREAD

**Issue by CorbanR** [ansible/ansible-modules-core#1083]
##### Issue Type:

Bug Report
##### Ansible Version:

`ansible --version
ansible 1.9.0.1`
`apache-libcloud (0.17.0)`
##### Environment:

`Mac OS X Yosemite: Version 10.10.2`
##### Summary:

Ansible gce - create or terminate GCE instances `metadata` is not working as expected. 

Example: 
This is valid

```
metadata: '{"sshKeys":"root:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDQK+dX349U1PV9qmMvj6DgZnnnTVfkLodRZ0MOBFSCzXYhN4KQMPuom/oJAPPYBpHEbaYOocZRTVIYuPO5u7gwXOx+oazmvhzpmjO6R/ZwiY8X4a07LonuxWhf2ry0cWAIVr06x+5AOuBBrJazdVxwhPYlJC4QEdMMAxtFB+uRU6nhSjPzxVGAbZSNTGvrlasDLcxkQLAsnETEFldopStW/t12udEm5nnBPNi2S9WG2eru5mPKPyug4WkqYxjdBRNCwnKUl5NV9r7GM25SxXpbjf72fFSRSCzOLe3eiC9z6/TFzlSmWjGZRNqGEcbKj3HdLBJZSSIOq3vYX0+EItg/"}'
```

This is failing and should be valid

```
metadata: '{"sshKeys":"root:{{ ssh_key.stdout }}"}'
```
##### Reproduce Failure:

**Example #1 Tasks:** 

```
- name: Get the default SSH key
  command: cat ~/.ssh/test_rsa.pub
  register: ssh_key

- debug: msg="{{ ssh_key.stdout }}"

- name: Set temp fact for troubleshooting purposes
  set_fact: temp='{"sshKeys":"root:{{ ssh_key.stdout }}"}'

- name: Set gce_meta_data fact for troubleshooting purposes
  set_fact: gce_meta_data='{{ temp }}'

- debug: var=gce_meta_data

- name: Create the GCE instance
  local_action:
    module: gce
    name: "{{ gce_server_name }}"
    network: "{{ gce_network }}"
    zone: "{{ gce_zone }}"
    machine_type: "{{ gce_machine_type }}"
    pem_file: "{{ gce_pem_file }}"
    project_id: "{{ gce_project_id }}"
    service_account_email: "{{ gce_service_account_email }}"
    image: "{{ gce_image }}"
    tags: "{{ gce_server_name }}"
    metadata: '{{ temp }}' 
```

**Output:** 

```
TASK: [genesis-google | Get the default SSH key] ******************************
changed: [127.0.0.1] => {"changed": true, "cmd": ["cat", "~/.ssh/test_rsa.pub"], "delta": "0:00:00.003559", "end": "2015-04-04 14:07:41.084497", "rc": 0, "start": "2015-04-04 14:07:41.080938", "stderr": "", "stdout": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDQK+dX349U1PV9qmMvj6DgZnnnTVfkLodRZ0MOBFSCzXYhN4KQMPuom/oJAPPYBpHEbaYOocZRTVIYuPO5u7gwXOx+oazmvhzpmjO6R/ZwiY8X4a07LonuxWhf2ry0cWAIVr06x+5AOuBBrJazdVxwhPYlJC4QEdMMAxtFB+uRU6nhSjPzxVGAbZSNTGvrlasDLcxkQLAsnETEFldopStW/t12udEm5nnBPNi2S9WG2eru5mPKPyug4WkqYxjdBRNCwnKUl5NV9r7GM25SxXpbjf72fFSRSCzOLe3eiC9z6/TFzlSmWjGZRNqGEcbKj3HdLBJZSSIOq3vYX0+EItg/", "warnings": []}

TASK: [genesis-google | debug msg="{{ ssh_key.stdout }}"] *********************
ok: [127.0.0.1] => {
    "msg": "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDQK+dX349U1PV9qmMvj6DgZnnnTVfkLodRZ0MOBFSCzXYhN4KQMPuom/oJAPPYBpHEbaYOocZRTVIYuPO5u7gwXOx+oazmvhzpmjO6R/ZwiY8X4a07LonuxWhf2ry0cWAIVr06x+5AOuBBrJazdVxwhPYlJC4QEdMMAxtFB+uRU6nhSjPzxVGAbZSNTGvrlasDLcxkQLAsnETEFldopStW/t12udEm5nnBPNi2S9WG2eru5mPKPyug4WkqYxjdBRNCwnKUl5NV9r7GM25SxXpbjf72fFSRSCzOLe3eiC9z6/TFzlSmWjGZRNqGEcbKj3HdLBJZSSIOq3vYX0+EItg/"
}

TASK: [genesis-google | Set temp fact for troubleshooting purposes] ***********
ok: [127.0.0.1] => {"ansible_facts": {"temp": "{\"sshKeys\":\"root:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDQK+dX349U1PV9qmMvj6DgZnnnTVfkLodRZ0MOBFSCzXYhN4KQMPuom/oJAPPYBpHEbaYOocZRTVIYuPO5u7gwXOx+oazmvhzpmjO6R/ZwiY8X4a07LonuxWhf2ry0cWAIVr06x+5AOuBBrJazdVxwhPYlJC4QEdMMAxtFB+uRU6nhSjPzxVGAbZSNTGvrlasDLcxkQLAsnETEFldopStW/t12udEm5nnBPNi2S9WG2eru5mPKPyug4WkqYxjdBRNCwnKUl5NV9r7GM25SxXpbjf72fFSRSCzOLe3eiC9z6/TFzlSmWjGZRNqGEcbKj3HdLBJZSSIOq3vYX0+EItg/\"}"}}

TASK: [genesis-google | Set gce_meta_data fact for troubleshooting purposes] ***
ok: [127.0.0.1] => {"ansible_facts": {"gce_meta_data": "{\"sshKeys\":\"root:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDQK+dX349U1PV9qmMvj6DgZnnnTVfkLodRZ0MOBFSCzXYhN4KQMPuom/oJAPPYBpHEbaYOocZRTVIYuPO5u7gwXOx+oazmvhzpmjO6R/ZwiY8X4a07LonuxWhf2ry0cWAIVr06x+5AOuBBrJazdVxwhPYlJC4QEdMMAxtFB+uRU6nhSjPzxVGAbZSNTGvrlasDLcxkQLAsnETEFldopStW/t12udEm5nnBPNi2S9WG2eru5mPKPyug4WkqYxjdBRNCwnKUl5NV9r7GM25SxXpbjf72fFSRSCzOLe3eiC9z6/TFzlSmWjGZRNqGEcbKj3HdLBJZSSIOq3vYX0+EItg/\"}"}}

TASK: [genesis-google | debug var=gce_meta_data] ******************************
ok: [127.0.0.1] => {
    "var": {
        "gce_meta_data": {
            "sshKeys": "root:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDQK+dX349U1PV9qmMvj6DgZnnnTVfkLodRZ0MOBFSCzXYhN4KQMPuom/oJAPPYBpHEbaYOocZRTVIYuPO5u7gwXOx+oazmvhzpmjO6R/ZwiY8X4a07LonuxWhf2ry0cWAIVr06x+5AOuBBrJazdVxwhPYlJC4QEdMMAxtFB+uRU6nhSjPzxVGAbZSNTGvrlasDLcxkQLAsnETEFldopStW/t12udEm5nnBPNi2S9WG2eru5mPKPyug4WkqYxjdBRNCwnKUl5NV9r7GM25SxXpbjf72fFSRSCzOLe3eiC9z6/TFzlSmWjGZRNqGEcbKj3HdLBJZSSIOq3vYX0+EItg/"
        }
    }
}

TASK: [genesis-google | Create the GCE instance] ******************************
failed: [127.0.0.1 -> 127.0.0.1] => {"failed": true, "parsed": false}
failed=True msg='bad metadata: malformed string'


FATAL: all hosts have already failed -- aborting
```
##### Reproduce Success:

**Example Tasks:** 

```
- name: Get the default SSH key
  command: cat ~/.ssh/test_rsa.pub
  register: ssh_key

- name: Create the GCE instance
  local_action:
    module: gce
    name: "{{ gce_server_name }}"
    network: "{{ gce_network }}"
    zone: "{{ gce_zone }}"
    machine_type: "{{ gce_machine_type }}"
    pem_file: "{{ gce_pem_file }}"
    project_id: "{{ gce_project_id }}"
    service_account_email: "{{ gce_service_account_email }}"
    image: "{{ gce_image }}"
    tags: "{{ gce_server_name }}"
    metadata: '{"sshKeys":"root:ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDQK+dX349U1PV9qmMvj6DgZnnnTVfkLodRZ0MOBFSCzXYhN4KQMPuom/oJAPPYBpHEbaYOocZRTVIYuPO5u7gwXOx+oazmvhzpmjO6R/ZwiY8X4a07LonuxWhf2ry0cWAIVr06x+5AOuBBrJazdVxwhPYlJC4QEdMMAxtFB+uRU6nhSjPzxVGAbZSNTGvrlasDLcxkQLAsnETEFldopStW/t12udEm5nnBPNi2S9WG2eru5mPKPyug4WkqYxjdBRNCwnKUl5NV9r7GM25SxXpbjf72fFSRSCzOLe3eiC9z6/TFzlSmWjGZRNqGEcbKj3HdLBJZSSIOq3vYX0+EItg/"}'
```
##### Expected Results:

I would have expected 

```
metadata: '{"sshKeys":"root:{{ ssh_key.stdout }}"}'
```

or in the example above 

```
metadata: '{{ temp }}'
```

to be valid. 
##### Actual Results:

```
failed=True msg='bad metadata: malformed string'
```

**Comment by CorbanR** [comment]
Implemented the fix suggested in https://github.com/ansible/ansible-modules-core/issues/154 and this seems to have fixed the issue for me. 

```
Change from:

    md = literal_eval(metadata)
To:

    md = literal_eval(str(metadata))
```

**Comment by bcoca** [comment]
cc @erjohnso

**Comment by erjohnso** [comment]
This was fixed in f620a0ac41c82edc1e894734f4e0c4dbf63dfd3f
